### DISCUSSION TITLE
Reproducing errors: how to serialize and deserialize manifold objects?

### DISCUSSION THREAD

**Discussion by Loosetooth** [elalish/manifold#485] [Q&A]
Let's say I have a few manifold objects, that when unioned together result in an error being thrown.

It would be nice if I could easily share these objects with the developers of manifold to try and reproduce the error, without having to share / reproduce the complete code generating those shapes.

I could try to export the GLB files, but then I'd have to be able to import them again too.
Is it currently possible to import them somehow into manifoldcad.org?

One idea I had would be to fork the manifold repo, and place the glb files in the `wasm` folder, so that the files could be available through the webserver. One could then easily look at the manifoldcad instance in my fork. But it's not clear to me how I could then 'import' those files in the file editor.

Any other ideas are welcome.

**Comment by elalish** [comment]
Great question, and one I've been contemplating as well. We can read and write GLBs losslessly now, see an example [here](https://manifoldcad.org/make-manifold). You could stand up your own website ([Glitch](https://glitch.com/) might be a simple choice) where you can upload your GLBs and have some JS that loads Manifold and does the problematic ops. Honestly, we should probably make a starter Glitch for people to remix to make this process easier. Want to give it a try?

**Reply by elalish** [reply]
I also just figured out how to debug our WASM in Chrome, which I added to our [README](https://github.com/elalish/manifold/blob/master/bindings/wasm/examples/README.md), which will make this process all the more useful.

**Reply by Loosetooth** [reply]
Now that you mentioned [the glb import/export example](https://manifoldcad.org/make-manifold), perhaps it makes sense to just make a modified version of this page. Let the user upload their GLB files, have a radio button selection for `union, difference, intersect`, and an 'execute' button. Then at least the boolean operations can easily be tested on provided GLB files.

**Reply by elalish** [reply]
Good idea! Sounds like a cross between that page and this one: https://manifoldcad.org/model-viewer.html

Would you like to try a PR for it?

**Reply by Loosetooth** [reply]
OK, I'll give it a shot.

**Reply by Loosetooth** [reply]
@elalish Where can I find the source code for [this page](https://manifoldcad.org/make-manifold)? I'm a bit lost.

**Reply by pca006132** [reply]
@Loosetooth https://github.com/elalish/manifold/blob/master/bindings/wasm/examples/make-manifold.html

**Reply by Loosetooth** [reply]
Ah thanks, I was unknowingly on a really old commit locally.
